### PR TITLE
Images: add IfixitImage component

### PR DESCRIPTION
**Pull request by danielbeardsley**
We want fine grained control over when and how we use next/image and its image resizing pipeline. Given that each of our images are rendered in many sizes, we want to be able to choose the sizes to use and when.

This is a minimal starting point. This gives us one place where we can tune when to use which sizes and when to use next's image pipeline. For the time being, all images are served directly from their sources instead of being re-compressed on the fly.

CC @sterlinghirsh @andyg0808

### CODE REVIEW

**Review by andyg0808 — APPROVED** [review]

**Comment by andyg0808** [comment]
CR :goat: through 296bedf

File: frontend/components/ifixit-image/ifixit-image.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+   if (typeof props.src === 'string') {
+      if (isGuideImage(props.src)) {
+         unoptimized = true;
+      } else if (isCartImage(props.src)) {
+         unoptimized = true;
+      } else if (isStrapiImage(props.src)) {
+         unoptimized = true;
+      }
+   }
```
Comment (andyg0808) [inline frontend/components/ifixit-image/ifixit-image.tsx:14]: Might consider pulling the conditional below out into its own function, which allows setting `unoptimized` once:
```suggestion
   const unoptimized = props.unoptimized || !shouldOptimize(props.src);
```

Example `shouldOptimize`:
```ts
function shouldOptimize(src: any) {
   if (typeof src === 'string') {
      if (isGuideImage(src)) {
         return false;
      } else if (isCartImage(src)) {
         return false;
      } else if (isStrapiImage(src)) {
         return false;
      }
   }
   return true;
}
```

**Review by andyg0808 — APPROVED** [review]

**Comment by andyg0808** [comment]
CR :goat: through 68505ab

File: frontend/components/ifixit-image/ifixit-image.tsx (hunk 1 above)
Comment (danielbeardsley) [reply frontend/components/ifixit-image/ifixit-image.tsx:14]: In upcoming pulls we'll want to change both the loader *and* `unoptimized`. When that comes, we'll probably user a pattern like `const [unoptimized, loader] = getLoader(props)`

**Comment by danielbeardsley** [comment]
Seems to be QA :+1: 

cdn and cominor.com images are used directly, while static images go through the pipeline.

image

**Review by andyg0808 — APPROVED** [review]

**Comment by andyg0808** [comment]
CR :goat: through c8003cd

**Pull request merged**